src/Sarif.Multitool/Rules/SARIF2009.UseConventionalSymbolicNames.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+using Microsoft.Json.Pointer;
+
+namespace Microsoft.CodeAnalysis.Sarif.Multitool.Rules
+{
+    public class UseConventionalSymbolicNames : SarifValidationSkimmerBase
+    {
+        /// <summary>
+        /// SARIF2009
+        /// </summary>
+        public override string Id => RuleId.UseConventionalSymbolicNames;
+
+        /// <summary>
+        /// Adopt uniform naming conventions for the symbolic names that SARIF uses various contexts.
+        /// 
+        /// Many tools follow a conventional format for the 'reportingDescriptor.id' property: a short string identifying the tool concatenated with a numeric rule number,
+        /// for example, 'CS2001' for a diagnostic from the Roslyn C# compiler. For uniformity of experience across tools, we recommend this format.
+        /// 
+        /// Many tool use similar names for 'uriBaseId' symbols.We suggest 'REPOROOT' for the root of a repository, 'SRCROOT' for the root of the directory containing all source code, 'TESTROOT' for the root of the directory containing all test code(if your repository is organized in that way), and 'BINROOT' for the root of the directory containing build output(if your project places all build output in a common directory).
+        /// </summary>
+        public override MultiformatMessageString FullDescription => new MultiformatMessageString { Text = RuleResources.SARIF2009_UseConventionalSymbolicNames_FullDescription_Text };
+
+        protected override IEnumerable<string> MessageResourceNames => new string[] {
+            nameof(RuleResources.SARIF2009_UseConventionalSymbolicNames_Warning_UseConventionalRuleIds_Text),
+            nameof(RuleResources.SARIF2009_UseConventionalSymbolicNames_Warning_UseConventionalUriBaseIdNames_Text)
+        };
+
+        public override FailureLevel DefaultLevel => FailureLevel.Warning;
```

Comment:
>Warning [](start = 66, length = 7)

Default level probably too high for this rule governing name conventions

Reply:
Just updated to Note.

---
In reply to: [445550154](https://github.com/microsoft/sarif-sdk/pull/1938#discussion_r445550154) [](ancestors = 445550154)